src/ITensorNetworks/MPScalculus/contract.jl

```diff
@@ -0,0 +1,54 @@
+using ChainRulesCore
+
+struct MPSTensor
+  mps::MPS
+  cutoff::Float64
+  maxdim::Integer
+end
+
+AbstractTensor = Union{ITensor,MPSTensor}
+
+function ITensors.contract(mps1::MPS, mps2::MPS; cutoff=1e-15, maxdim=1000)
+  ## TODO: modify this function based on https://arxiv.org/pdf/1912.03014.pdf
+  tensor = contract(vcat(mps1.data, mps2.data)...)
+  return if size(tensor) == ()
+    MPS([tensor])
+  else
+    MPS(tensor, inds(tensor); cutoff=cutoff, maxdim=maxdim)
+  end
+end
+
+MPSTensor(mps::MPS) = MPSTensor(mps, 1e-15, 1000)
+
+function ITensors.inds(tensor::MPSTensor)
+  return siteinds(tensor.mps) == [nothing] ? [] : siteinds(tensor.mps)
+end
+
+# contract into one ITensor
+ITensors.ITensor(t::MPSTensor) = contract(t.mps.data)
+
+ITensors.contract(t1::MPSTensor) = t1
+
+function ITensors.contract(t1::MPSTensor, t2::MPSTensor)
+  mps_out = contract(t1.mps, t2.mps; cutoff=t1.cutoff, maxdim=t1.maxdim)
+  return MPSTensor(mps_out, t1.cutoff, t1.maxdim)
+end
+
+ITensors.contract(t1::MPSTensor, t2::MPSTensor...) = contract(t1, contract(t2...))
+
+ITensors.contract(t_list::Vector{MPSTensor}) = contract(t_list...)
+
+function Base.getindex(t::MPSTensor)
+  @assert length(t.mps.data) == 1
+  @assert size(t.mps.data[1]) == ()
```

Comment:
Maybe instead use `@assert order(only(t.mps)) == 0` (the `only(t.mps)` call checks if it is exactly length 1, so then the check above won't be needed).

Reply:
Seems to work fine for me:
```julia
(@v1.6) pkg> st ITensors
      Status `~/.julia/environments/v1.6/Project.toml`
  [9136182c] ITensors v0.2.5

julia> using ITensors

julia> M = MPS([ITensor(1.3)])
MPS
[1] ITensor()


julia> only(M)[]
1.3
```

Reply:
Maybe you are using an older version of Julia?

Reply:
`only` was introduced in Julia v1.4 (https://github.com/JuliaLang/julia/blob/master/HISTORY.md#new-library-functions-3).

Reply:
But anyway, if it is causing trouble then checking `length(t.mps) == 1` and then extracting the tensor as `t.mps[1]` is fine, I just like `only` since it does the check and extracts the value in a single call.

Reply:
That's wired:
`julia
(v1.3) pkg> st ITensors
    Status `~/.julia/environments/v1.3/Project.toml`
  [9136182c] ITensors v0.2.3

julia> using ITensors

julia> M = MPS([ITensor(1.3)])
MPS
[1] ITensor()


julia> only(M)[]
ERROR: UndefVarError: only not defined
Stacktrace:
 [1] top-level scope at REPL[8]:1
`

Reply:
I guess julia 1.3 doesn't support `only` yet?

Reply:
Oh I see, yeah I was using Julia v1.3. Will just not use this for now.